File: back-python/calculator.py
"""Logique métier de la calculatrice (équivalent du Calculator Node.js)."""


class Calculator:
    def add(self, a, b):
        return a + b

    def subtract(self, a, b):
        return a - b

    def multiply(self, a, b):
        return a * b

    def divide(self, a, b):
        if b == 0:
            raise ValueError("Division par zéro impossible.")
        return a / b

File: back-java/src/main/java/com/calculator/Calculator.java
package com.calculator;

/** Logique métier de la calculatrice (équivalent du Calculator Node.js). */
public class Calculator {

    public double add(double a, double b) {
        return a + b;
    }

    public double subtract(double a, double b) {
        return a - b;
    }

    public double multiply(double a, double b) {
        return a * b;
    }

    public double divide(double a, double b) {
        if (b == 0) {
            throw new ArithmeticException("Division par zéro impossible.");
        }
        return a / b;
    }
}
